core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -99,6 +118,9 @@ class TiBatchWrite(@transient val df: DataFrame,
     tiTableRef = options.tiTableRef
     tiDBInfo = tiSession.getCatalog.getDatabase(tiTableRef.databaseName)
     tiTableInfo = tiSession.getCatalog.getTable(tiTableRef.databaseName, tiTableRef.tableName)
+    uniqueIndices = tiTableInfo.getIndices.asScala.filter(index => index.isUnique).toList
+    handleCol = tiTableInfo.getPrimaryKeyColumn
```

Comment:
handleCol -> primaryKeyCol?

Reply:
handleCol means pk is handle column. Primary key col just means primary key column. The previous one has more restricted meaning.

Reply:
`handleCol` may not equals to `tiTableInfo.getPrimaryKeyColumn` in some cases, e.g. table with char type primary key

Reply:
When it comes to this case, it is pk is not handle case. `handleCol` should be null.

Reply:
ok, now i know the problem.
the name `TiTableInfo.getPrimaryKeyColumn` should be changed to `TiTableInfo.getHandleColumn`.